app/controllers/unattended_controller.rb

```diff
@@ -50,7 +55,6 @@ def hostgroup_template
   # Generate an action for each template kind
   # i.e. /unattended/provision will render the provisioning template for the requesting host
   def host_template
-    return head(:not_found) unless params[:kind].present?
```

Comment:
Its better to have this here rather than in a before action. Explicitness is king :)

Reply:
I agree, but this needs to happen before the method is called (that's why I had to move it to a before action).